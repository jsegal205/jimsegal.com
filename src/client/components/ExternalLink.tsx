import React, { FunctionComponent, ReactNode } from "react";

type ExternalLinkProps = {
  children: ReactNode;
  link: string;
  title: string;
};

const ExternalLink: FunctionComponent<ExternalLinkProps> = ({
  children,
  link,
  title,
}: ExternalLinkProps) => (
  <a href={link} rel="noopener noreferrer" target="_blank" title={title}>
    {children}
  </a>
);

export default ExternalLink;
