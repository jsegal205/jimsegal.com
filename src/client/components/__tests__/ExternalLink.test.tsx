import * as React from "react";
import { cleanup, render } from "@testing-library/react";

import ExternalLink from "../ExternalLink";

afterEach(cleanup);

test("renders correct structure with text children", () => {
  const { container } = render(
    <ExternalLink link="#" title="title-prop">
      Link text
    </ExternalLink>
  );
  expect(container).toMatchSnapshot();
});

test("renders correct structure with HTML children", () => {
  const { container } = render(
    <ExternalLink link="#" title="title-prop">
      <div>inner div</div>
    </ExternalLink>
  );
  expect(container).toMatchSnapshot();
});
