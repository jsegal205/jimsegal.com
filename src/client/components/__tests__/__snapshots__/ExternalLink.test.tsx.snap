// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct structure with HTML children 1`] = `
<div>
  <a
    href="#"
    rel="noopener noreferrer"
    target="_blank"
    title="title-prop"
  >
    <div>
      inner div
    </div>
  </a>
</div>
`;

exports[`renders correct structure with text children 1`] = `
<div>
  <a
    href="#"
    rel="noopener noreferrer"
    target="_blank"
    title="title-prop"
  >
    Link text
  </a>
</div>
`;
